python/samples/getting_started/workflows/orchestration/handoff_agents.py

```diff
@@ -0,0 +1,337 @@
+    for message in request.conversation:
+        speaker = message.author_name or message.role.value
+        print(f"- {speaker}: {message.text}")
+    print("============================")
+
+
+async def main() -> None:
+    """Main entry point for the handoff workflow demo.
+
+    This function demonstrates:
+    1. Creating triage and specialist agents
+    2. Building a handoff workflow with custom termination condition
+    3. Running the workflow with scripted user responses
+    4. Processing events and handling user input requests
+
+    The workflow uses scripted responses instead of interactive input to make
+    the demo reproducible and testable. In a production application, you would
+    replace the scripted_responses with actual user input collection.
+    """
+    # Initialize the Azure OpenAI chat client
+    chat_client = AzureOpenAIChatClient(credential=AzureCliCredential())
+
+    # Create all agents: triage + specialists
+    triage, refund, order, support = create_agents(chat_client)
+
+    # Build the handoff workflow
+    # - participants: All agents that can participate (triage MUST be first or explicitly set as starting_agent)
+    # - starting_agent: The triage agent receives all user input first
+    # - with_termination_condition: Custom logic to stop the request/response loop
+    #   Default is 10 user messages; here we terminate after 4 to match our scripted demo
+    workflow = (
+        HandoffBuilder(
+            name="customer_support_handoff",
+            participants=[triage, refund, order, support],
+        )
+        .starting_agent("triage_agent")
+        .with_termination_condition(
+            # Terminate after 4 user messages (initial + 3 scripted responses)
+            # Count only USER role messages to avoid counting agent responses
+            lambda conv: sum(1 for msg in conv if msg.role.value == "user") >= 4
+        )
+        .build()
+    )
+
+    # Scripted user responses for reproducible demo
+    # In a real application, replace this with:
+    #   user_input = input("Your response: ")
+    # or integrate with a UI/chat interface
+    scripted_responses = [
+        "My order 1234 arrived damaged and the packaging was destroyed.",
+        "Yes, I'd like a refund if that's possible.",
+        "Thanks for resolving this.",
+    ]
+
+    # Start the workflow with the initial user message
+    # run_stream() returns an async iterator of WorkflowEvent
+    print("\n[Starting workflow with initial user message...]")
+    events = await _drain(workflow.run_stream("Hello, I need assistance with my recent purchase."))
+    pending_requests = _handle_events(events)
+
+    # Process the request/response cycle
+    # The workflow will continue requesting input until:
+    # 1. The termination condition is met (4 user messages in this case), OR
+    # 2. We run out of scripted responses
+    response_index = 0
+
+    while pending_requests and response_index < len(scripted_responses):
```

Comment:
nit, use `scripted_responses.pop()` and `while scripted_responses` for more OOP style code.